src/EllipsisNotation.jl

```diff
@@ -3,16 +3,45 @@ __precompile__()
 module EllipsisNotation
 
   import Base.getindex, Base.setindex!
-  const ..   = Val{:..}
+  const .. = Val{:..}()
+  const IndexType = Union{(VERSION < v"0.6-" ? Real : Int), Val{:..}}
```

Comment:
It looks like it was a change in Base:

https://github.com/JuliaLang/julia/blob/v0.5.2/base/array.jl#L385
https://github.com/JuliaLang/julia/blob/master/base/array.jl#L514

The array interface says to use `getindex(A, I::Vararg{Int, N})`, but it's clear that these definitions are going to cause problems on `Array` types in v0.5. Specifically it looks like this commit made everything more succinct:

https://github.com/JuliaLang/julia/commit/8a55211f3b9c20ea2baedabbecd56ff9f4286284

and gets rid of  the ambiguity errors. So what I am going to do is add that change back in, release it as a patch, and then when I go to v0.6 drop this. But this seems necessary for v0.5. I'm sorry I didn't check this more thoroughly before.